### ISSUE TITLE
ed448 signature malleability

### ISSUE THREAD

**Issue by reaperhulk** [openssl/openssl#7706]
Similar to #7693, ed448 verification doesn't check that `s` is less than the group order, allowing malleability for some signatures. Wycheproof doesn't have vectors for this yet, but I generated an example:

```c
#include <assert.h>
#include <stdio.h>
#include <string.h>
#include <openssl/crypto.h>
#include <openssl/evp.h>
#include <openssl/x509.h>


int main(void) {
    OPENSSL_init();
    int ret = 0;
    int res;
    unsigned char *public = OPENSSL_hexstr2buf("fedb02a658d74990244d9d10cf338e977565cbbda6b24c716829ed6ee1e4f28cf2620c052db8d878f6243bffc22242816c1aaa67d2f3603600", NULL);
    unsigned char *sig = OPENSSL_hexstr2buf("0cc16ba24d69277f927c1554b0f08a2a711bbdd20b058ccc660d00ca13542a3cef9e5c44c54ab23a2eb14f947e167b990b080863e28b399380f30db6e54d5d1406d23378ffde11b1fb81b2b438a3b8e8aa7f7f4e1befcc905023fab5a5465053844f04cf0c1b51d84760f869588687f57500", NULL);
    unsigned char *msg = OPENSSL_hexstr2buf("38", NULL);

    EVP_PKEY *pkey = EVP_PKEY_new_raw_public_key(NID_ED448, NULL, public, 57);
    assert(pkey != NULL);
    EVP_MD_CTX *ctx = EVP_MD_CTX_new();
    res = EVP_DigestVerifyInit(ctx, NULL, NULL, NULL, pkey);
    assert(res == 1);
    res = EVP_DigestVerify(ctx, sig, 114, msg, 1);
    if (res == 1) {
        printf("Verification passed but should have failed.\n");
        ret = 1;
    }
    exit(ret);
}
```

**Comment by mattcaswell** [comment]
> Either that or I'm doing something stupid

Yep. Me being stupid.

Apparently even though `s` is defined to be 57 bytes long by the standard, the order `L` is in fact only 56 bytes long! So a valid value for `s` always ends in `0x00`, and if `s >=L` then the two most significant bytes must be `0x003F` or greater. Which it is in your test case (`0x0075`)

**Comment by mattcaswell** [comment]
> Can you provide the equivalent valid sig?

Forget that...no longer needed.

**Comment by mattcaswell** [comment]
Fix in #7748.
